pym/bob/cmds/build.py

```diff
@@ -1429,20 +1438,35 @@ def doProject(argv, bobRoot):
     if args.preserve_env: extra.append('-E')
     if args.sandbox: extra.append('--sandbox')
 
-    package = packages.walkPackagePath(args.package)
 
-    # execute a bob dev with the extra arguments to build all executables.
+    backlog = []
+    for p in args.packages:
+        for package in packages.queryPackagePath(p):
+            backlog.append(package)
+
+    # execute a bob dev/build with the extra arguments to build all executables.
     # This makes it possible for the plugin to collect them and generate some runTargets.
     if args.execute_prebuild:
         devArgs = extra.copy()
         if args.resume: devArgs.append('--resume')
         if args.execute_buildonly: devArgs.append('-b')
-        devArgs.append(args.package)
-        doDevelop(devArgs, bobRoot)
+        for p in backlog:
+            devArgs.append(p.getName())
+        if args.build:
+            doBuild(devArgs, bobRoot)
+        else:
+            doDevelop(devArgs, bobRoot)
+
+    if len(backlog) == 1:
```

Comment:
This is not a good idea. Depending on the number of given packages the plugin will receive an `Package` or a list of `Package`s. Either pass an object (old plugin API < 0.13) or a list (new plugin API >= 0.13).

Reply:
Yeah, as mentioned I also think this is a bad idea. Will always provide a list then.